pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -843,11 +850,15 @@ fn wal_stream_connection_string(
             let host = url.host_str().context("host is missing")?;
             let port = url.port().unwrap_or(5432); // default PG port
 
-            Ok(format!(
+            let no_secrets = format!(
                 "host={host} \
                  port={port} \
                  options='-c timeline_id={timeline_id} tenant_id={tenant_id}'"
-            ))
+            );
+            Ok(ConnectionString {
+                no_secrets: no_secrets.clone(),
+                with_secrets: no_secrets,
+            })
```

Comment:
It looks like at the moment there's no real difference between `no_secrets` and `with_secrets`. I'd advise to stick with YAGNI and add another field once we introduce at least one secret that shouldn't be printed. Until then, you could add helper TODO/XXX/NOTE comments with the reasoning.